cmd/puma-dev/command.go

```diff
@@ -15,11 +19,65 @@ func command() error {
 	switch flag.Arg(0) {
 	case "link":
 		return link()
+	case "status":
+		return status()
 	default:
 		return fmt.Errorf("Unknown command: %s\n", flag.Arg(0))
 	}
 }
 
+type App struct {
+	Status  string `json:"status"`
+	Scheme  string `json:"scheme"`
+	Address string `json:"address"`
+}
+
+func status() error {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", "http://localhost/status", nil)
```

Comment:
does puma-dev always expose status on port 80? if you're running on a nonstandard port, you'd need to access it elsewhere. 

admittedly, :80 is very likely the most common case. but, it might be worth respecting `fHTTPPort` when checking the status so `puma-dev -http-port 8080 status` would work.

Reply:
Possibly, but the status API documentation makes no mention of needing to use a different port other than `:80` here. You can just run `curl -H "Host: puma-dev" localhost/status`, and as long as `localhost` resolves to `puma-dev`'s DNS server, it should work.

Reply:
we could consider the dynamic port a future enhancement. i think this will get the job done for the majority of users.

eventually, we'll need to have dynamic port in order to call this feature complete. if, say, you're not interested in running puma-dev as a daemon on 80, you won't be able to use `status` but maybe that's okay, as long as the error message communicates that. 

Here's a quick example of checking status of two different puma-dev's running.

```shell
osiris:~$ curl -H "Host: puma-dev" localhost/status # get status of puma-dev running as daemon
{}
osiris:~$ curl -H "Host: puma-dev" localhost:9280/status # get status of puma-dev running interactively
{}
osiris:~$ curl -H "Host: epimetheus" localhost:9280/ | head -n1 # boot an app via interactive puma-dev
<!DOCTYPE html>
osiris:~$ curl -H "Host: puma-dev" localhost:9280/status | head -n1 # get status of puma-dev running interactively, now that an app has been booted
{"epimetheus":{"scheme":".......}
osiris:~$ curl -H "Host: puma-dev" localhost/status # get status of puma-dev running as daemon
{}
osiris:~$

```

Reply:
That's interesting...are you running on macOS or Linux? I'm getting a different response when I try to `curl` the status using the default HTTP port number on macOS Catalina:

```
$ curl -H "Host: puma-dev" localhost:9280/status
curl: (7) Failed to connect to localhost port 9280: Connection refused
```

Some other info of note:

```
$ launchctl list | grep io.puma 
27687   0       io.puma.dev
```

```
$ ps -ef | grep puma-dev
  502 27687     1   0 11:17AM ??         0:00.36 /usr/local/Cellar/puma-dev/0.12/bin/puma-dev -launchd -dir ~/.puma-dev -d test -timeout 15m0s
  502 77111 76171   0 12:21PM ttys005    0:00.47 vim cmd/puma-dev/command.go
  502 77879 76171   0 12:22PM ttys005    0:00.00 grep puma-dev
```

Also, of course:

```
$ curl -H "Host: puma-dev" localhost/status
{}
```

This may be due to a file (which was apparently generated by puma-dev at some point) in **/etc/resolver/localhost**:

```
# Generated by puma-dev
nameserver 127.0.0.1
port 9253
```

Reply:
i'm sorry. i didn't include enough context. i'm _explicitly_ running puma-dev in interactive mode alongside my daemonized puma-dev on macOS in order to prove that `/status` listens on whatever port puma-dev is configured to use.

```
osiris:puma-dev(cover-homedir-report-coverage)$ puma-dev
* Directory for apps: /Users/norton/.puma-dev
* Domains: test
* DNS Server port: 9253
* HTTP Server port: 9280
* HTTPS Server port: 9283
! Puma dev listening on http and https
! Booting app 'epimetheus' on socket /Users/norton/.puma-dev/epimetheus/tmp/puma-dev-63200.sock
epimetheus[63609]: bash: no job control in this shell
epimetheus[63609]: Puma starting in single mode...
epimetheus[63609]: * Version 4.3.0 (ruby 2.6.5-p114), codename: Mysterious Traveller
epimetheus[63609]: * Min threads: 0, max threads: 5
epimetheus[63609]: * Environment: development
epimetheus[63609]: * Listening on unix:/Users/norton/.puma-dev/epimetheus/tmp/puma-dev-63200.sock
epimetheus[63609]: Use Ctrl-C to stop
! App 'epimetheus' booted
! Killing 'epimetheus' (63609)
epimetheus[63609]: - Gracefully stopping, waiting for requests to finish
epimetheus[63609]: === puma shutdown: 2020-02-14 11:41:54 -0500 ===
epimetheus[63609]: - Goodbye!
! Killing 'epimetheus' (63609)
* App 'epimetheus' shutdown and cleaned up
```